plugins/verification/src/main/java/org/qbicc/plugin/verification/ClassInitializingBasicBlockBuilder.java

```diff
@@ -0,0 +1,76 @@
+package org.qbicc.plugin.verification;
+
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.BasicBlockBuilder;
+import org.qbicc.graph.DelegatingBasicBlockBuilder;
+import org.qbicc.graph.Value;
+import org.qbicc.graph.ValueHandle;
+import org.qbicc.type.ClassObjectType;
+import org.qbicc.type.definition.DefinedTypeDefinition;
+import org.qbicc.type.definition.element.ConstructorElement;
+import org.qbicc.type.definition.element.FieldElement;
+import org.qbicc.type.definition.element.InitializerElement;
+import org.qbicc.type.definition.element.MethodElement;
+
+/**
+ *
+ */
+public class ClassInitializingBasicBlockBuilder extends DelegatingBasicBlockBuilder {
+
+    private final CompilationContext ctxt;
+
+    public ClassInitializingBasicBlockBuilder(final CompilationContext ctxt, final BasicBlockBuilder delegate) {
+        super(delegate);
+        this.ctxt = ctxt;
+    }
+
+    @Override
+    public ValueHandle instanceFieldOf(ValueHandle instance, FieldElement field) {
+        initialize(field.getEnclosingType());
+        return super.instanceFieldOf(instance, field);
+    }
+
+    @Override
+    public ValueHandle staticField(FieldElement field) {
+        initialize(field.getEnclosingType());
+        return super.staticField(field);
+    }
+
+    @Override
+    public ValueHandle exactMethodOf(Value instance, MethodElement method) {
+        initialize(method.getEnclosingType());
+        return super.exactMethodOf(instance, method);
+    }
+
+    @Override
+    public ValueHandle virtualMethodOf(Value instance, MethodElement method) {
+        initialize(method.getEnclosingType());
+        return super.virtualMethodOf(instance, method);
+    }
+
+    @Override
+    public ValueHandle staticMethod(MethodElement method) {
+        initialize(method.getEnclosingType());
+        return super.staticMethod(method);
+    }
+
+    @Override
+    public ValueHandle constructorOf(Value instance, ConstructorElement constructor) {
+        initialize(constructor.getEnclosingType());
+        return super.constructorOf(instance, constructor);
+    }
+
+    @Override
+    public Value new_(ClassObjectType type) {
+        initialize(type.getDefinition());
+        return super.new_(type);
+    }
+
+    private void initialize(DefinedTypeDefinition definition) {
+        InitializerElement initializer = definition.load().getInitializer();
+        if (initializer != null) {
+            ctxt.enqueue(initializer);
```

Comment:
This is insufficient as it misses all the super class and interface initializers.  See https://github.com/qbicc/qbicc/pull/530/commits/7f668e6c9c52be4a359b335a3bbb9205839ae7f0 for details on which initializers need to be enqueued

Reply:
Right, I didn't set out to solve that problem (yet) - to start I wanted to solve the immediate problem of constants being defined too late.  But I think this approach could easily be extended to cover those cases.

Reply:
I've pushed up a modification to initialize the superclass of any classes used.  I *think* that covers all the cases that we need to cover.

Reply:
There's also the interfaces that define default methods that need also need to be initialized per the spec.

This code, copied from ReachabilityBlockBuilder, would cover the interfaces:
```
// Walk all interfaces implemented by the current class and its supers
            // and enqueue their <clinit> if they have a default method
            ltd.forEachInterfaceFullImplementedSet(i -> {
                if (i.declaresDefaultMethods()) {
                    InitializerElement initializer = i.getInitializer();
                    if (initializer != null) {
                        ctxt.enqueue(initializer);
                    }
                }
            });
```

Reply:
Ah, I figured I probably missed a case. I was thinking we only needed to init interfaces for static methods, and didn't even think of default methods.

Reply:
I updated for this case, but I had to do it a little bit differently because `initialize` is already recursive.

Reply:
Looks right to me